packages/amplify-cli-core/src/amplify-node-pkg-detector/package-lock-parser.ts

```diff
@@ -0,0 +1,106 @@
+import _ from 'lodash';
+import { AmplifyFault } from '../errors/amplify-fault';
+import { JSONUtilities } from '../jsonUtilities';
+import { LockfileType } from './lock-file-types';
+
+/**
+ * package lock interface
+ */
+export interface PackageLock {
+    name: string;
+    version: string;
+    dependencies?: PackageLockDependency;
+    lockfileVersion: 1 | 2;
+    type: LockfileType.NPM;
+}
+
+/**
+ * package lock dependency interface
+ */
+export interface PackageLockDependency {
+    [depName: string]: PackageLockDependencyType;
+}
+
+/**
+ * package lock dependencies type
+ */
+export interface PackageLockDependencyType {
+    version: string;
+    requires?: {
+      [depName: string]: string;
+    };
+    dependencies?: PackageLockDependency;
+    dev?: boolean;
+}
+
+/**
+ * package lock parser
+ */
+export class PackageLockParser {
+  type: LockfileType;
+  dependenciesMap:Record<string, Record<string, PackageLockDependencyType>>;
```

Comment:
why do we need this as a class attribute?

Reply:
this stores all packages present in lockfiles which are dependent on `dependencyToSearch`(i.e. @aws-cdk/core in cdk case)

Reply:
I meant why specifically a _class attribute_. This seems like an output field which we can create in `getDependentPackageMap` and return it from there to avoid it getting manipulated by some other method in future.

Reply:
I am manipulating this is class methods so I thought to keep as as class private variable.